js/portfolio/articles.js

```diff
@@ -1,54 +1,87 @@
-var articles = [];
-
 function Article (opts) {
-  this.ttl = opts.ttl;
-  this.category = opts.category;
-  this.author = opts.author;
-  this.authorUrl = opts.authorUrl;
-  this.img = opts.img;
-  this.pubDate = opts.pubDate;
-  this.bdy = opts.bdy;
-  this.year = opts.year;
+  for (keys in opts) {
+    this[keys] = opts[keys];
+  }
 }
 
-Article.prototype.toHtml = function() {
+Article.all = [];
+
+Article.prototype.toHtml = function(scriptTemplateId) {
+  var template = Handlebars.compile((scriptTemplateId).html());
 
-  // this.year = new Date(this.pubDate).getFullYear();
-  // console.log(this.year);
-  this.daysAgo = parseInt((new Date() - new Date(this.pubDate))/60/60/24/1000);
+  this.daysAgo = parseInt((new Date() - new Date(this.pubDate)) / 60 / 60 / 24 / 1000);
   this.publishStatus = this.pubDate ? 'published about ' + this.daysAgo + ' days ago' : '(draft)';
-  var $source = $('#article-template').html();
-  var template = Handlebars.compile($source);
-  return template(this);
-};
 
-Article.prototype.populateDateFilter = function(){
-  var $source = $('#date-filter-template').html();
-  var template = Handlebars.compile($source);
   return template(this);
 };
 
-Article.prototype.populateCategoryFilter = function(){
-  var $source = $('#category-filter-template').html();
-  var template = Handlebars.compile($source);
-  return template(this);
+Article.loadAll = function(inputData) {
+  inputData.sort(function(a, b) {
+    return (new Date(b.pubDate)) - (new Date(a.pubDate));
+  });
+  inputData.forEach(function(ele) {
+    Article.all.push(new Article(ele));
+  });
 };
 
-myPortfolioData.sort(function(a,b) {
-  return (new Date(b.pubDate)) - (new Date(a.pubDate));
-});
+Article.fetchAll = function() {
+  if (localStorage.eTagValue) {
+    var newETag;
 
-myPortfolioData.forEach(function(ele) {
-  articles.push(new Article(ele));
-});
+    console.log('localStorage etag value is ' + JSON.parse(localStorage.eTagValue));
 
-articles.forEach(function(a){
-  $('#articles').append(a.toHtml());
-  $('#date-filter').append(a.populateDateFilter());
-  $('#category-filter').append(a.populateCategoryFilter());
+    $.ajax({
+      url: 'data/portfolioData.json',
+      type: 'HEAD',
+      success: function(data, message, xhr) {
+        newETag = xhr.getResponseHeader('eTag');
+        console.log('new eTag is ' + newETag);
+        console.log('stored etag is ' + JSON.parse(localStorage.eTagValue));
 
-  $('.filter option').each(function(){
-    $(this).siblings('[value="' + this.value + '"]').remove();
-  });
+        if (newETag === JSON.parse(localStorage.eTagValue)) {
+          console.log('no change');
+          Article.loadAll(JSON.parse(localStorage.portfolioData));
+          portfolioView.initIndexPage();
+
+        } else {
+          $.ajax({
```

Comment:
Is there a way you can DRY out this and the later AJAX call that's very similar?